cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,342 @@
+  cy.getElementByTestId('advancedSelectorLanguageSelect').select(datasetLanguage);
+  cy.getElementByTestId('advancedSelectorTimeFieldSelect').select(
+    "I don't want to use the time filter"
+  );
+  cy.getElementByTestId('advancedSelectorConfirmButton').click();
+}
+
+/**
+ * Select an index dataset.
+ * @param {string} indexClusterName Name of the cluster to be used for the Index.
+ * @param {string} indexName Name of the index dataset to be used.
+ * @param {string} datasetLanguage Index supports "OpenSearch SQL" and "PPL".
+ */
+export function selectIndexDataset(indexClusterName, indexName, datasetLanguage) {
+  cy.getElementByTestId('datasetSelectorButton').click();
+  cy.getElementByTestId('datasetSelectorAdvancedButton').click();
+  cy.getElementByTestId('datasetExplorerWindow').contains('Indexes').click();
+  cy.getElementByTestId('datasetExplorerWindow')
+    .contains(indexClusterName, { timeout: 10000 })
+    .click();
+  cy.getElementByTestId('datasetExplorerWindow').contains(indexName, { timeout: 10000 }).click();
+  cy.getElementByTestId('datasetSelectorNext').click();
+  selectIndexDatasetLanguage(datasetLanguage);
+}
+
+/**
+ * Select a language in the Dataset Selector for Index Pattern
+ * @param {string} datasetLanguage Index Pattern supports "DQL", "Lucene", "OpenSearch SQL" and "PPL"
+ */
+export function selectIndexPatternDatasetLanguage(datasetLanguage) {
+  cy.getElementByTestId('advancedSelectorLanguageSelect').select(datasetLanguage);
+  cy.getElementByTestId('advancedSelectorConfirmButton').click();
+}
+
+/**
+ * Select an index pattern dataset.
+ * @param {string} indexPatternName Name of the index pattern to be used.
+ * @param {string} datasetLanguage Index Pattern supports "DQL", "Lucene", "OpenSearch SQL" and "PPL"
+ */
+export function selectIndexPatternDataset(indexPatternName, datasetLanguage) {
+  cy.getElementByTestId('datasetSelectorButton').click();
+  cy.getElementByTestId('datasetSelectorAdvancedButton').click();
+  cy.getElementByTestId('datasetExplorerWindow').contains('Index Patterns').click();
+  cy.getElementByTestId('datasetExplorerWindow')
+    .contains(indexPatternName, { timeout: 10000 })
+    .click();
+  cy.getElementByTestId('datasetSelectorNext').click();
+  selectIndexPatternDatasetLanguage(datasetLanguage);
+}
+
+/**
+ * Toggle expansion of row rowNumber of Doc Table.
+ * @param {number} rowNumber rowNumber of Doc Table starts at 0 for row 1.
+ */
+export function toggleDocTableRow(rowNumber) {
+  getDocTableRow(rowNumber).within(() => {
+    cy.getElementByTestId('docTableExpandToggleColumn').find('button').click();
+  });
+}
+
+/**
+ * Check the query hit text matches expectedQueryHitText.
+ * @param {string} expectedQueryHitsText expected text for query hits. Commas must be added e.g. 10,000
+ */
+export function verifyQueryHitsText(expectedQueryHitsText) {
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsText);
+}
```

Comment:
You should be able to use this: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/cypress/utils/apps/data_explorer/commands.js#L44 (although looking at it now, i wonder if it accounts for the `commas`. If not i think we should update that one to account for commas

Reply:
Changed to use that. I'm not going to make that change in the current pr, but I do agree. It should be taking an integer as well, instead of a string that you have to manually add commas.